starboard/tools/testing/test_coverage.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/python
+#
+# Copyright 2023 The Cobalt Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Generate coverage reports from llvm coverage data."""
+
+import argparse
+import glob
+import logging
+import os
+import pathlib
+import subprocess
+
+from starboard.build import clang
+from starboard.tools import build
+from starboard.tools.util import SetupDefaultLoggingConfig
+
+SRC_DIR = os.path.join(
+    os.path.dirname(__file__), os.pardir, os.pardir, os.pardir)
+
+
+def _get_raw_reports(test_binaries, coverage_dir):
+  """Get a list of raw coverage reports from the coverage directory."""
+  raw_reports = []
+  missing_reports = []
+  for target in test_binaries:
+    # Find coverage reports from the supplied test binaries.
+    profraw_file = os.path.join(coverage_dir, target + '.profraw')
+    if not os.path.isfile(profraw_file):
+      missing_reports.append(profraw_file)
+    else:
+      raw_reports.append(profraw_file)
+  else:
+    # If no test binaries were supplied we use all coverage data we find.
+    raw_reports = glob.glob(os.path.join(coverage_dir, '*.profraw'))
+
+  if missing_reports:
+    logging.error('Unable to find the following coverage files: %s.',
+                  ', '.join(missing_reports))
+  return raw_reports
+
+
+def _get_test_binary_paths(binaries_dir, test_binaries, coverage_dir):
+  if not test_binaries:
+    # Use the coverage files to deduce the paths if no binaries were specified.
+    raw_reports = _get_raw_reports(test_binaries, coverage_dir)
+    test_binaries = [pathlib.Path(report).stem for report in raw_reports]
+  return [os.path.join(binaries_dir, binary) for binary in test_binaries]
+
+
+def _get_exclude_opts():
+  """Get list of exclude options."""
+  # Get a list of all directories in the src root.
+  all_dirs = next(os.walk(SRC_DIR))[1]
+  # TODO(oxv): Make configurable?
```

Comment:
nit - TODO seems unnecessary. Simple for the moment, and easy to extend when needed.